lti_consumer/plugin/compat.py

```diff
@@ -19,3 +21,47 @@ def run_xblock_handler_noauth(*args, **kwargs):
     # pylint: disable=import-error,import-outside-toplevel
     from lms.djangoapps.courseware.module_render import handle_xblock_callback_noauth
     return handle_xblock_callback_noauth(*args, **kwargs)
+
+
+def get_user_from_external_user_id(external_user_id):
+    """
+    Import ExternalId model and find user by external_user_id
+    """
+    # pylint: disable=import-error,import-outside-toplevel
+    from openedx.core.djangoapps.external_user_ids.models import ExternalId
+    try:
+        external_id = ExternalId.objects.get(
+            external_user_id=external_user_id,
+            external_id_type__name='lti'
+        )
+        return external_id.user
+    except ExternalId.DoesNotExist as exception:
+        raise LtiError('Invalid User') from exception
+    except ValidationError as exception:
+        raise LtiError('Invalid userID') from exception
+
+
+def submit_grade(score):
```

Comment:
Move all logic from here to the signal handler, this function should have as little logic as possible.
Make this function only import `grades_signals` and trigger the signal, and receive any required input as parameters:
```suggestion
def publish_grade(block, user, score, possible):
```

You can ignore the other parameters/leave them empty for now.

Also, when moving the logic, make sure to check if the submission is late and it the block allows a late grade.

Reply:
@giovannicimolin Refactored. Kept other parameters as ``None`` now and moved all logic to signal.